src/Chapter4.hs

```diff
@@ -629,7 +664,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM m1 m2 = m1 >>= (\x -> if x then m2 else pure x)
```

Comment:
Thank you for reviewing, it's really a great project you guys have here 😄 

I originally had `andM m1 m2 = m1 >>= (\x -> m2 >>= (\y -> pure (x && y)))` but this failed the test for `Just False|Nothing`. I'm wondering if this is more accurate since `andM (Just False) Nothing` would return `Nothing` with my first implementation

Reply:
Thank you for your kind words! :blush: 

The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: